jobs/kubelet/templates/bin/drain.erb

```diff
@@ -70,6 +63,7 @@ check_if_pidfile_exists() {
 }
 
 drain_kubelet() {
+  local node_name=$(kubectl --kubeconfig /var/vcap/jobs/kubeconfig/config/kubeconfig get nodes -o wide | grep "<%= spec.ip %>" | awk '{print $1}')
```

Comment:
Oh I see, because it ignores the hostname override on vsphere when the cloud provider is passed in.